src/server/scripts/EasternKingdoms/zone_north_sea.cpp

```diff
@@ -0,0 +1,119 @@
+/*
+ * This file is part of the TrinityCore Project. See AUTHORS file for Copyright information
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "Conversation.h"
+#include "GridNotifiersImpl.h"
+#include "Map.h"
+#include "ScriptMgr.h"
+#include "ScriptedEscortAI.h"
+
+enum CaptainGarrickConversation
+{
+    CONVERSATION_WARMING_UP = 12818,
+};
+
+enum CaptainGarrickQuests
+{
+    QUEST_WARMING_UP = 56775,
+};
+
+enum CaptainGarrick
+{
+    CREATURE_GARRICK_QUARTERDECK_SPAWN_ID = 1050589,
+};
+
+struct npc_captain_garrick : public EscortAI
+{
+    npc_captain_garrick(Creature* creature) : EscortAI(creature) { }
+
+    const float CloneOrientation = 5.124503135681152343f;
+    const float CloneSpawnZOffSet = 0.308f;
+
+    void OnQuestAccept(Player* player, Quest const* quest) override
+    {
+        if (quest->GetQuestId() != QUEST_WARMING_UP)
+            return;
+
+        Conversation::CreateConversation(CONVERSATION_WARMING_UP, player, *player, player->GetGUID());
+    }
+
+    void OnQuestReward(Player* player, Quest const* quest, LootItemType /*type*/, uint32 /*opt*/) override
+    {
+        if (quest->GetQuestId() != QUEST_WARMING_UP)
+            return;
+
+        Creature* garrick = me->GetMap()->GetCreatureBySpawnId(CREATURE_GARRICK_QUARTERDECK_SPAWN_ID);
+        if (!garrick)
+            return;
+
+        Position pos(me->GetPositionX(), me->GetPositionY(), me->GetPositionZ() - CloneSpawnZOffSet, CloneOrientation);
+        garrick->SummonPersonalClone(pos, TEMPSUMMON_MANUAL_DESPAWN, 0s, 0, 0, player, me->GetGUID());
+    }
+
+    void InitializeAI() override
+    {
+        EscortAI::InitializeAI();
+
+        if (!me->IsPrivateObject())
+            return;
+
+        SetPauseTimer(1ms);
+        me->RemoveNpcFlag(UNIT_NPC_FLAG_QUESTGIVER);
+    }
+
+    void IsSummonedBy(WorldObject* /*owner*/) override
+    {
+        _scheduler.Schedule(1600ms, [this](TaskContext task)
+        {
+            Talk(0);
```

Comment:
magic number

Reply:
its not like TALK_0 or 0 will make any difference here tbh

Reply:
doesn't matter. magic numbers are magic numbers. get rid of it

Reply:
> doesn't matter. magic numbers are magic numbers. get rid of it

these names for TALK_xx are not even a bit better than writing 0, 1, ... kek.